vcs/gitcmd/repo.go

```diff
@@ -197,6 +197,20 @@ func (r *Repository) Branches(opt vcs.BranchesOptions) ([]*vcs.Branch, error) {
 		}
 	}
 
+	if opt.ContainsCommit != "" {
+		filteredBranchNames, err := r.branchesWithCommit(opt.ContainsCommit)
+		if err != nil {
+			return nil, err
+		}
+		filteredBranches := make([]*vcs.Branch, 0)
+		for _, branch := range branches {
+			if filteredBranchNames[branch.Name] {
```

Comment:
I personally prefer making presence checks explicit:

```go
if _, present := filteredBranchNames[branch.Name]; present {
```

That way, you don't have to think, "Can the map value ever be false?"

Then you make the map (and the return type of `branchesWithCommit`) a `map[string]struct{}`, which makes it clear that the return value is a set, not a map.

There are memory savings too, to using a map value type of `struct{}`, I believe, but those matter less.